hcs-sxc-java-core/src/main/java/com/hedera/hcs/sxc/callback/OnHCSMessageCallback.java

```diff
@@ -53,30 +78,59 @@ public OnHCSMessageCallback (HCSCore hcsCore) throws Exception {
      * Adds an observer to the list of observers
      * @param listener
      */
+    @Override
     public void addObserver(HCSCallBackToAppInterface listener) {
        observers.add(listener);
     }
     /**
      * Notifies all observers with the supplied message
      * @param message
+     * @param applicationMessageId
      */
+    @Override
     public void notifyObservers(byte[] message, ApplicationMessageId applicationMessageId) {
         HCSResponse hcsResponse = new HCSResponse();
         hcsResponse.setApplicationMessageId(applicationMessageId);
+    
+        
         hcsResponse.setMessage(message);
         observers.forEach(listener -> listener.onMessage(hcsResponse));
     }
+    
+    @Override
     public void storeMirrorResponse(SxcConsensusMessage consensusMessage) {
         hcsCore.getMessagePersistence().storeMirrorResponse(consensusMessage);
     }
-    public void partialMessage(ApplicationMessageChunk messagePart) throws InvalidProtocolBufferException {
+    @Override
+    public void partialMessage(ApplicationMessageChunk messagePart) throws InvalidProtocolBufferException{
 
         Optional<ApplicationMessage> messageEnvelopeOptional =
                 pushUntilCompleteMessage(messagePart, this.hcsCore.getMessagePersistence());
 
-        if (messageEnvelopeOptional.isPresent()){
+        if (messageEnvelopeOptional.isPresent()){ // is present if all parts received
             this.hcsCore.getMessagePersistence().storeApplicationMessage(messageEnvelopeOptional.get().getApplicationMessageId(), messageEnvelopeOptional.get());
-            notifyObservers( messageEnvelopeOptional.get().toByteArray(), messageEnvelopeOptional.get().getApplicationMessageId());
+                
+            ApplicationMessage appMessage = messageEnvelopeOptional.get();
+            
+            if(this.encryptMessages){

```

Comment:
This flag is to indicate messages should be encrypted, there is an override on message encryption (this boolean could be false and a message be encrypted, the opposite is true too), we should have a flag in the application message that indicates whether the message was encrypted or not by the sender and use that here to determine if decryption is necessary (maybe raise a separate issue and assume appnet-wide encryption for now)

Reply:
See issue #371

Reply:
See issue #371